File: Python3/프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.py
def solution(a, b, c):
    s1, s2, s3 =  (a + b + c), (pow(a, 2) + pow(b, 2) + pow(c, 2)) , (pow(a, 3) + pow(b, 3) + pow(c, 3))
    
    if a == b and b == c:
        return s1 * s2 * s3
    elif a == b or b == c or a == c:
        return s1 * s2
    return s1

File: C++/프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int a, int b, int c) {
    int answer = a + b + c;
    if(a == b || b == c || a == c) { //둘 중 하나 같음 보장
        answer *= a * a + b * b + c * c;
    }
    if(a == b && b == c) { //셋 모두 같음 보장
        answer *= a * a * a + b * b * b + c * c * c;
    }
    
    return answer;
}